app/models/ems_refresh/save_inventory.rb

```diff
@@ -389,4 +374,32 @@ def save_ems_storage_inventory(ems, hashes, target = nil)
     save_ems_block_storage_inventory(ems, hashes, target) if ems.supports?(:block_storage)
     save_ems_object_storage_inventory(ems, hashes, target) if ems.supports?(:object_storage)
   end
+
+  def find_vm(h, vms_by_uid_ems)
+    # Find the Vm in the database with the current uid_ems.  In the event
+    #   of duplicates, try to determine which one is correct.
+    duplicates = vms_by_uid_ems[h[:uid_ems]] || []
+
+    # Reject any VMs which are active on other EMSs
+    duplicates.reject! { |vm| vm.ems_id.present? && vm.ems_id != h[:ems_id] }
```

Comment:
I vaguely remember the original code needed to *not* modify the vms_by_uid_ems...can't remember why, but I'm thinking if this is a pure refactor, maybe we shouldn't change this?

Reply:
Okay done, I think adding `:ems_id => [nil, ems.id]` to the `VmOrTemplate.where(:uid_ems => vms_uids)` query above would still be a good addition to reduce the possible VMs that are going to be implicitly filtered out later on but we can do this later

Reply:
Since we found that we aren't "stealing" VMs from other EMSs anymore, I changed the original query to explicitly exclude VMs from other EMSs: https://github.com/ManageIQ/manageiq/pull/18625/files#diff-090b07f911cee85c2acff15e6bda1032R58